OlinkAnalyze/vignettes/bridging_introduction.Rmd

```diff
@@ -0,0 +1,233 @@
+  fig.width = 10,
+  message = FALSE,
+  warning = FALSE,
+  time_it = TRUE,
+  fig.width = 7
+)
+```
+
+
+Olink® NPX datasets are normalized datasets using either plate control
+normalization or intensity normalization methods. Intensity
+normalization method assumes that all samples within a project are fully
+randomized.
+
+The joint analysis of two or more Olink® NPX datasets often requires
+additional batch correction step to remove technical variations, which
+is referred to as bridging.
+
+Bridging is needed if Olink® NPX datasets are:
+
+-   plate control normalized only and run conditions (e.g lab and
+    reagent lots) have changes.
+
+-   intensity normalized but from two different sample populations.
+
+To bridge two or more Olink® NPX datasets, bridging samples are needed
+to calculate adjustment factors between datasets. Bridging samples are 
+overlapping samples between datasets. The recommended number
+of bridging samples for Explore 1536 datasets is between 8-16. Olink® 
+NPX datasets without overlapping samples can not be combined to perform 
+joint analysis using the bridging approach described below.
+
+The following tutorial is designed to give you an overview of the kinds
+of data combining methods that are possible using the Olink® bridging
+procedure.
+
+## Setup bridging datasets
+
+```{r setup}
+library(OlinkAnalyze)
+library(dplyr)
+library(stringr)
+library(ggplot2)
+library(kableExtra)
+```
+
+Bridging datasets are standard Olink® NPX tables. They can be loaded using 
+`read_NPX()` function with NPX manager output file as input.
+
+```{r message=FALSE, eval=FALSE, echo = TRUE}
+data1 <- read_NPX("~/NPX_file1_location.xlsx")
+data2 <- read_NPX("~/NPX_file2_location.xlsx")
+```  
+
+To demonstrate how bridging works, we will use the example datasets
+(`npx_data1` and `npx_data2`) from **OlinkAnalyze** package.
+
+## Check bridging datasets
+
+Explore and have an overview of the datasets that are going to be bridged. For 
+example, plot and compare NPX distribution between datasets.
+
+```{r check, message=FALSE}
+# Load datasets
+npx_1 <- npx_data1 %>% 
+  filter(!str_detect(SampleID, 'CONTROL_SAMPLE')) %>% #Remove control samples
+  mutate(dataset = "data1")
+npx_2 <- npx_data2 %>% 
+  filter(!str_detect(SampleID, 'CONTROL_SAMPLE')) %>% #Remove control samples
+  mutate(dataset = "data2")
+
+npx_df <- bind_rows(npx_1,npx_2)
+
+
+# Plot NPX density before bridging normalization
+npx_df %>% 
+  mutate(Panel = gsub("Olink ", "", Panel)) %>% 
+  ggplot(aes(x = NPX, fill = Panel)) +
+  geom_density(scale = 1, alpha = 0.7) + 
+  facet_grid(dataset~Panel) +
+  olink_fill_discrete(coloroption = c('teal', 'lightblue')) +
+  set_plot_theme() + 
+  ggtitle("Before bridging normalization: NPX distribution") +
+  theme(axis.title.x = element_blank(),
+        axis.title.y = element_blank(),
+        strip.text = element_text(size=16),
+        legend.title = element_blank(),
+        legend.position = "none")
+```  
+
```

Comment:
I prefer the overlaid density plots for this as I think it more readily highlights differences.